File: webServicreditos/ClientApp/src/app/servicreditos/models/credito.ts
export class Credito {
    identificacion: string;
    nombre: string;
    capitalInicial: number;
    tasaInteres: number;
    tiempoDuracion: number;

    calcularTasaInteres(){
        return this.tasaInteres / 100;
    }

    calcularTotalPagar(){
        if(this.capitalInicial > 0 && this.tiempoDuracion > 0){
            return this.capitalInicial * (Math.pow(1 + this.calcularTasaInteres(), this.tiempoDuracion));
        }
        return 0;
    }
}

File: Entity/Credito.cs
using System;

namespace Entity
{
    public class Credito
    {
        public string Identificacion { get; set; }
        public string Nombre { get; set; }
        public decimal CapitalInicial { get; set; }
        public int TasaInteres { get; set; }
        public int TiempoDuracion { get; set; }
        public int TotalPagar { get; set; }

        public decimal CalcularTasaInteres(){
            return this.TasaInteres / 100;
        }

        public decimal CalcularTotalPagar(){
            if(this.CapitalInicial > 0 && this.TiempoDuracion > 0){
                this.TotalPagar = this.CapitalInicial * (Math.Pow(1 + this.CalcularTasaInteres(), this.TiempoDuracion));
                return this.TotalPagar;
            }
            return 0;
        }

    }
}
